app/components/ConnectButton.tsx

```diff
@@ -1,93 +1,76 @@
 import { Button } from "~/components/button";
 
-function CustomConnectButton() {
+function useSanctionScreening() {
   const { disconnect } = useDisconnect();
+  const account = useAccount();
 
-  const [connected, setConnected] = useState(false);
-
-  const account = useAccount({
-    onConnect({ address }) {
-      logger.info(address + " successfully connected");
-      setConnected(true);
-    },
-  });
-
-  const { data: isSanctionedAddress } = useChainalysisContract(account.address ? account.address : "");
+  const { data: isSanctionedAddress } = useChainalysisContract(account.address ?? "");
 
   useEffect(() => {
-    if (isSanctionedAddress === true && connected) {
+    if (isSanctionedAddress === true) {
       disconnect();
-      setConnected(false);
-      logger.info("address " + account.address + " was sanctioned and therefore disconnected");
+      alert("address " + account.address + " was sanctioned and therefore disconnected");
     }
-  }, [connected, disconnect, account.address, isSanctionedAddress]);
+  }, [disconnect, account.address, isSanctionedAddress]);
+}
+
+// https://www.rainbowkit.com/docs/custom-connect-button
+function CustomConnectButton() {
+  useSanctionScreening();
 
   return (
     <ConnectButton.Custom>
       {({ account, chain, openAccountModal, openChainModal, openConnectModal, authenticationStatus, mounted }) => {
-        if (isSanctionedAddress === true) {
-          authenticationStatus = "unauthenticated";
-        }
-
         const ready = mounted && authenticationStatus !== "loading";
         const connected =
           ready && account && chain && (!authenticationStatus || authenticationStatus === "authenticated");
 
-        return (
-          <div
-            {...(!ready && {
-              "aria-hidden": true,
-              style: {
-                opacity: 0,
-                pointerEvents: "none",
-                userSelect: "none",
-              },
-            })}
-          >
-            {(() => {
-              if (authenticationStatus === "unauthenticated") {
-                return <Button onClick={openConnectModal}>Sanctioned Address</Button>;
-              } else if (!connected) {
-                return (
-                  <Button variant="gradient" onClick={openConnectModal}>
-                    Connect Wallet
-                  </Button>
-                );
-              }
+        if (!ready) {
+          // The docs make this a blank state with opactiy 0, but might be better to show a loading indicator of some sort
+          return (
+            <Button variant="gradient" className="pointer-events-none select-none opacity-0">
+              Connect Wallet
+            </Button>
+          );
+        }
 
-              if (chain.unsupported || chain.name == "Ethereum") {
-                return (
-                  <Button variant="danger" onClick={openChainModal}>
-                    Switch to Polygon
```

Comment:
I know they wanted this to specifically say "Switch to Polygon" for now, doesn't matter that much though